src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/Tokenization.cs

```diff
@@ -0,0 +1,320 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.PowerFx.Core.Binding;
+using Microsoft.PowerFx.Core.Binding.BindInfo;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Intellisense;
+using Microsoft.PowerFx.Syntax;
+
+namespace Microsoft.PowerFx.Core.Texl.Intellisense
+{
+    /// <summary>
+    /// An entity with methods to compute ordered or unordered enumeration enumeration of token text spans in a expression rule with their start and end indices and token type.
+    /// </summary>
+    internal static class Tokenization
+    {
+        /// <summary>
+        /// Returns an ordered enumeration of token text spans in a expression rule with their start and end indices and token type. Orders token by default comparer.
+        /// </summary>
+        /// <param name="expression">Expression which would be tokenized.</param>
+        /// <param name="binding">Binding which would be used to tokenize the operation and determine the type of each token.</param>
+        /// <param name="comments">Colllection of comment tokens extracted from the given expression.</param>
+        /// <returns>Ordered collection of tokens.</returns>
+        internal static ICollection<ITokenTextSpan> Tokenize(string expression, TexlBinding binding, IEnumerable<CommentToken> comments = null)
+        {
+            return Tokenize(expression, binding, comments, new TokenTextSpanComparer(), true);
+        }
+
+        /// <summary>
+        /// Returns an enumeration of token text spans in a expression rule with their start and end indices and token type.
+        /// </summary>
+        /// <param name="expression">Expression which would be tokenized.</param>
+        /// <param name="binding">Binding which would be used to tokenize the operation and determine the type of each token.</param>
+        /// <param name="comments">Colllection of comment tokens extracted from the given expression.</param>
+        /// <param name="comparer">optional comparer to sort tokens.</param>
+        /// <param name="computeHiddenTokens">Optional flag to indicate whether to compute whether token can be hidden or not.</param>
+        /// <returns>Ordered or undordered collection of tokens.</returns>
+        internal static ICollection<ITokenTextSpan> Tokenize(string expression, TexlBinding binding, IEnumerable<CommentToken> comments = null, IComparer<ITokenTextSpan> comparer = null, bool computeHiddenTokens = false)
```

Comment:
We don't want to compute hiddenness of the tokens for the new formula bar as we are not going to support that so by default it is off. However, we want to keep this for old formula bar as long as it exists and once we consume the changes in Canvas App backend, we would update the tokenization logic behind old formula bar to use this one instead and computeHiddenTokens would be true for old formula bar.

Reply:
Consider also adding part of this as a comment for this arg, capture that we don't intend to enable it at any point.